### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by happyCupcake**

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
 
 stdenv.mkDerivation {
   name = "drivebrain_software";
   src = ./.;
   nativeBuildInputs = [ cmake pkg-config ];
-  propagatedBuildInputs = [ protobuf lz4 zstd boost cmake_macros nlohmann_json foxglove-ws-protocol-cpp hytech_np_proto_cpp dbcppp gtest drivebrain_core_msgs_proto_cpp mcap db_service_grpc_cpp grpc vn_lib matlab_math];
+  propagatedBuildInputs = [ protobuf lz4 zstd boost cmake_macros nlohmann_json foxglove-ws-protocol-cpp hytech_np_proto_cpp dbcppp gtest drivebrain_core_msgs_proto_cpp mcap db_service_grpc_cpp grpc vn_lib matlab_math spdlog fmt];
+
+cmakeFlags = [
+    "-DCMAKE_PREFIX_PATH=${fmt}/lib/cmake;${spdlog}/lib/cmake"
```
Comment (RCMast3r) [inline default.nix:10]: you shouldnt need to do this, nix already adds these for you

File: drivebrain_app/main.cpp [hunk 2]
```diff
@@ -43,7 +47,8 @@ std::atomic<bool> stop_signal{false};
 // Signal handler function
 void signalHandler(int signal)
 {
-    std::cout << "Interrupt signal (" << signal << ") received. Cleaning up..." << std::endl;
+    // std::cout << "Interrupt signal (" << signal << ") received. Cleaning up..." << std::endl;
```
Comment (RCMast3r) [inline drivebrain_app/main.cpp:50]: go ahead and remove all the places you commented the logs out

File: drivebrain_app/main.cpp [hunk 3]
```diff
@@ -82,7 +87,10 @@ int main(int argc, char *argv[])
     po::notify(vm);
     
     if (vm.count("help")) {
-        std::cout << desc << "\n";
+        // std::cout << desc << "\n";
```
Comment (RCMast3r) [inline drivebrain_app/main.cpp:90]: pls remove commented code

File: drivebrain_app/main.cpp [hunk 4]
```diff
@@ -137,26 +145,31 @@ int main(int argc, char *argv[])
     DBInterfaceImpl db_service_inst(message_logger);
     std::thread db_service_thread([&db_service_inst]()
                                   {
-            std::cout <<"started db service thread" <<std::endl;
+            // std::cout <<"started db service thread" <<std::endl;
```
Comment (RCMast3r) [inline drivebrain_app/main.cpp:148]: commented code

File: drivebrain_app/main.cpp [hunk 5]
```diff
@@ -137,26 +145,31 @@ int main(int argc, char *argv[])
...
                                   {
-            std::cout <<"started io context thread" <<std::endl;
+            // std::cout <<"started io context thread" <<std::endl;
+            spdlog::info("Started io context thread");
             try {
                 io_context.run();
             } catch (const std::exception& e) {
-                std::cerr << "Error in io_context: " << e.what() << std::endl;
+                //std::cerr << "Error in io_context: " << e.what() << std::endl;
+                spdlog::error("Error in io_context: {}", e.what());
```
Comment (RCMast3r) [inline drivebrain_app/main.cpp:172]: commented code

File: drivebrain_app/main.cpp [hunk 6]
```diff
@@ -243,11 +256,13 @@ int main(int argc, char *argv[])
...
-    std::cout << "joined main process" << std::endl;
+    // std::cout << "joined main process" << std::endl;
+    spdlog::info("joined main process");
     io_context.stop();
     io_context_thread.join();
     db_service_inst.stop_server();
     db_service_thread.join();
-    std::cout << "joined io context" << std::endl;
+    // std::cout << "joined io context" << std::endl;
+    spdlog::info("joined io context");
```
Comment (RCMast3r) [inline drivebrain_app/main.cpp:266]: commented code

File: drivebrain_core_base/src/JsonFileHandler.cpp
Comment (RCMast3r) [inline drivebrain_core_base/src/JsonFileHandler.cpp:1]: commented code

**Review by RCMast3r — CHANGES_REQUESTED** [review]
looks good, except go ahead and remove all the commented out code, no need to keep it around and fix the nix derivation.

**Review by RCMast3r — APPROVED** [review]

**Review by SreeDan — APPROVED** [review]

**Pull request merged**